packages/shortest/src/core/framework-detector/index.ts

```diff
@@ -0,0 +1,81 @@
+import { existsSync } from "fs";
+import fs from "fs/promises";
+import path from "path";
+import { listFrameworks } from "@netlify/framework-info";
+import { simpleGit, SimpleGit, CleanOptions } from "simple-git";
+import { DOT_SHORTEST_DIR_NAME, DOT_SHORTEST_DIR_PATH } from "@/cache";
+import { getLogger } from "@/log";
+import { directoryExists } from "@/utils/directory-exists";
+import { getErrorDetails, ShortestError } from "@/utils/errors";
+
+const PROJECT_JSON_PATH = path.join(DOT_SHORTEST_DIR_PATH, "project.json");
+
+export const detectFramework = async (options: { force?: boolean } = {}) => {
+  const log = getLogger();
+
+  if (!options.force && existsSync(PROJECT_JSON_PATH)) {
+    try {
+      const projectInfo = JSON.parse(
+        await fs.readFile(PROJECT_JSON_PATH, "utf-8"),
+      );
+      log.trace("Using cached framework information");
+      return projectInfo;
+    } catch (error) {
+      log.trace(
+        "Failed to read cached project data, performing detection",
+        getErrorDetails(error),
+      );
+    }
+  }
+
+  const frameworks = await listFrameworks({ projectDir: process.cwd() });
+
+  if (!(await directoryExists(DOT_SHORTEST_DIR_PATH))) {
+    await fs.mkdir(DOT_SHORTEST_DIR_PATH, { recursive: true });
+    log.trace(`Created ${DOT_SHORTEST_DIR_NAME} directory`);
+  }
+
+  try {
+    const VERSION = 1;
+
+    const projectInfo = {
+      metadata: {
+        timestamp: Date.now(),
+        version: VERSION,
+        git: await getGitInfo(),
```

Comment:
This could be used to invalidate cache in the future, with a simple check against current branch / last commit.